README.md

```diff
@@ -684,7 +721,82 @@ usr/src/uts/intel/os/name_to_major
+<https://smartos.org/bugview>. The commit message should be of the form used by
+illumos-gate, for example:
+
+```
+smartos-live#9999 make some changes
+Reviewed-by: Steve Reviewer <steve.reviewer@gmail.com>
+Approved-by: Amy Approver <amy.approver@gmail.com>
+```
+
+(Yes, there is no blank line after the commit title.)
+
+If you would like to make a change to `illumos-joyent` specifically, please see
+[Upstreaming](#upstreaming) below.
+
+## Review
+
+In general, before putting something up for review, some amount of
+testing should have already been done. Once you post it for review, then
+you need to seek out reviewers. A good first step for finding reviewers
+is to see who has worked on changes in similar areas. A good way to do
+this is to use `git log` in portions of the source tree and note who the
+authors, reviewers, and approvers have been. This can often be a good
+source of trying to figure out who to ask.
+
+If you're not sure of who to ask or are having trouble finding someone,
+then consider asking in a public forum such as internal chat or IRC.
+Even if you're not sure if someone would make sense as a reviewer or
+not, don't hesitate to reach out and folks will help you find or suggest
+reviewers. For more information on where to reach out, see the community
+section.
+
+## Upstreaming
+
+If you are making a change to `illumos-joyent`, please consider contributing
+directly to [illumos-gate](https://github.com/illumos/illumos-gate) instead.
+We automatically merge this into `illumos-joyent` every working day, so your fix
+will soon make it into SmartOS itself.
+
+The default case should be contributing directly to upstream. However, in areas
+of significant divergence, such as `lx` brand or certain areas of the networking
+stack, this may not be the best choice.
+
+## Integration
+
+When thinking about integrating, the following are questions that you or
+your approver should be asking yourself:
+
+* Have I tested this in all the ways I can think of? Might this impact
+standalone SmartOS or Triton in some way?
+* Have I documented any new commands or interfaces in manual pages?
+* Have I built this both debug and non-debug?
+* Have I reviewed the `git pbchk` output when working in bldenv in
+illumos-joyent?
+* Have I run any appropriate make check targets?
+* Have I looked for memory leaks?
+* Have I performed appropriate stress testing to try and find issues
+that might only arise after prolonged use?
+* Is this a particularly risky change? If so, should I wait
+until the start of the next release cycle to put this back?
+* Are there any heads up notices I need to send as part of this? For
+example, this might happen because of a flag day.
+* Have I added a new tool that's required to run at build-time and
+tested this on older platform images?
+
+Prior to a PR being merged, it must have at least one code reviewer and one
+approver. They can be the same person, but two sets of eyes are preferred.
```

Comment:
This should (eventually?) include a description of the "integration-approval" GitHub label process for getting an "approver"?